replica/replica.go

```diff
@@ -736,19 +747,23 @@ func (r *Replica) encodeToFile(obj interface{}, file string) error {
 		logrus.Errorf("failed to create temp file: %s while encoding the data to file", file)
 		return err
 	}
+
 	defer f.Close()
 
 	if err := json.NewEncoder(f).Encode(&obj); err != nil {
 		logrus.Errorf("failed to encode the data to file: %s", f.Name())
 		return err
 	}
 
-	if err := f.Close(); err != nil {
-		logrus.Errorf("failed to close file after encoding to file: %s", f.Name())
+	if err := r.closeAndSyncDir(f); err != nil {
```

Comment:
one doubt: already 'f' is closed in closeAndSyncDir but defer does close again.. isn't this some issue?